Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1818,26 +1818,37 @@ def json_to_str(self, json_input, is_pretty=True):
                 ['{key}: {val}'.format(key=k, val=flattenCell(v, is_pretty)) for k, v in json_input.items()])
 
         str_lst = []
-        for k, v in json_input.items():
-            str_lst.append('***{k}***: '.format(k=k))
-            if isinstance(v, dict):
-                items_to_add = ["\t**{key}**: {val}".format(key=key, val=flattenCell(val, is_pretty)) for key, val in
-                                self.item_generator(v)]
-            else:
-                items_to_add = ["\t{val}".format(val=flattenCell(v, is_pretty))]
-            str_lst.extend(items_to_add)
+        prev_path = None
+        for path, key, val in self.item_generator(json_input):
+            if path != prev_path:  # need to construct tha `path` string only of it changed from the last one
+                str_path = '\n'.join(["{tabs}**{p}**:".format(p=p, tabs=i * '\t') for i, p in enumerate(path)])
+                str_lst.append(str_path)
+                prev_path = path
+
+            str_lst.append(
+                '{tabs}***{key}***: {val}'.format(tabs=len(path) * '\t', key=key, val=flattenCell(val, is_pretty)))
+
         return '\n'.join(str_lst)
 
-    def item_generator(self, json_input):
+    def item_generator(self, json_input, path=None):
```

Comment:
Please add docstring + change the name to be more reflective.
perhaps `_json_to_path_generator`.